sigar_linux.go

```diff
@@ -720,6 +723,35 @@ func (self *ProcTime) Get(pid int) error {
 	self.StartTime += system.btime
 	self.StartTime *= 1000
 
+	return err
+}
+
+// Diff the counters to calculate percent of CPU usage. The "other" ProcTime should be from an earlier reading.
+func (self *ProcTime) Diff(other *ProcTime) error {
+	// The diffs need to be protected against underflow
+	if other.User > self.User {
+		return errors.New("Failed diff user ProcTime: operation would result in underflow")
+	}
+	if other.Sys > self.Sys {
+		return errors.New("Failed diff sys ProcTime: operation would result in underflow")
+	}
+	if other.CollectionTime.After(self.CollectionTime) {
+		return errors.New("'Other' ProcTime should be older than current")
+	}
+
+	diffUser := uint64(self.User - other.User)
+	diffSys := uint64(self.Sys - other.Sys)
+	diffMillis := uint64(self.CollectionTime.Sub(other.CollectionTime) / time.Millisecond)
+
+	if diffMillis == 0 {
+		return errors.New("Failed to diff ProcTime: operation would result in divide by zero")
+	}
+
+	// Calculate percentages, multiplying by 100 first to avoid precision loss
+	self.PercentUserTime = (diffUser * 100) / diffMillis
+	self.PercentSysTime = (diffSys * 100) / diffMillis
+	self.PercentTotalTime = ((diffUser + diffSys) * 100) / diffMillis
```

Comment:
This [doc](https://www.centos.org/docs/5/html/5.2/Deployment_Guide/s2-proc-stat.html) about `/proc/stat` says that time values are measured in 1/100ths of a second (jiffies).  Is that correction already done when creating the ProcTime, or does it need to be handled here?